cocotb_build_libs.py

```diff
@@ -19,6 +19,56 @@
 cocotb_share_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), "cocotb", "share"))
 
 
+def create_rc_file(name, filename, libraries):
+    archtitecture = "amd64" if sys.maxsize > 2**32 else "x86"
+    dependencies = []
+
+    for lib in libraries:
+        dependencies.append(
+            '        "<dependency>"\n'
+            '            "<dependentAssembly>"\n'
+            '                "<assemblyIdentity name=""%s"" version=""1.0.0.0"" type=""win32"" processorArchitecture=""%s"" />"\n'
+            '            "</dependentAssembly>"\n'
+            '        "</dependency>"\n' % (lib, archtitecture))
+
+    rc_body = (
+        '#include <WinUser.h>\n'
+        '\n'
+        'LANGUAGE 0x00, 0x00\n'
+        '\n'
+        'ISOLATIONAWARE_MANIFEST_RESOURCE_ID RT_MANIFEST\n'
+        'BEGIN\n'
+        '    "<?xml version=""1.0"" encoding=""UTF-8"" standalone=""yes""?>"\n'
+        '    "<assembly xmlns=""urn:schemas-microsoft-com:asm.v1"" manifestVersion=""1.0"">"\n'
+        '        "<assemblyIdentity name=""%s"" version=""1.0.0.0"" type=""win32"" processorArchitecture=""%s"" />"\n'
+        '        "<file name=""%s"" />"\n'
+        '%s'
+        '    "</assembly>"\n'
+        'END\n' % (name, archtitecture, filename, "".join(dependencies)))
+
+    with open(name + ".rc", "w", encoding='ascii') as f:
+        f.write(rc_body)
+
+
+def create_appconfig(filename):
+    config_body = (
+        '<?xml version="1.0" encoding="UTF-8" standalone="yes"?>\n'
+        '<configuration>\n'
+        '    <windows>\n'
+        '        <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">\n'
+        '            <probing privatePath="libs" />\n'
+        '        </assemblyBinding>\n'
+        '    </windows>\n'
+        '</configuration>\n')
+
+    dirpath = os.path.dirname(filename)
+    if dirpath and not os.path.exists(dirpath):
+        os.makedirs(dirpath)
+
+    with open(filename + ".2.config", "w", encoding='ascii') as f:
```

Comment:
Is it possible for a user non-ascii path component to end up in here?

Reply:
It should not be possible because the paths are only relative and therefore only contain `libs` but I changed it to UTF-8 as it doesn't hurt either.